### PR TITLE
Add HNPWA - Next.js HN implementation

### PR DESCRIPTION
**Pull request by chrisdwheatley**
Here's my implementation using the awesome Next.js.

### CODE REVIEW

File: site/_apps/nextjs-hn.txt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+api: 'Node-hnapi (unofficial)'
+hosting: 'Digital Ocean & Cloudflare'
+authors:
+    - name: 'Chris'
+lighthouse: '91/100'
+interactive-em: '4.7s'
+interactive-faster-3g: '3.0s'
+lighthouse-link: https://www.webpagetest.org/result/170607_PW_1d454612c736a65ddff058cf16b8842e/
+wpt-em-link: https://www.webpagetest.org/result/170607_PW_1d454612c736a65ddff058cf16b8842e/
+wpt-faster-3g-link: https://www.webpagetest.org/result/170607_DY_400d1cc5db11ec9727404dc9cbc441e6/
```
Comment (addyosmani) [inline site/_apps/nextjs-hn.txt]: Could we re-run this one? We generally read the first interactive score from the attached Lighthouse report (viewable by clicking through on the Lighthouse score). This ensures that the scores noted in an entry also correspond to what scores you got during the run. 

image

Attempting to kick off a re-run here: https://www.webpagetest.org/result/170607_S2_17b6c30a3f2f8fb964d12dfeb5dba8a0/

File: site/_apps/nextjs-hn.txt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+module-bundling: 'Next.js with additional Webpack config'
+service-worker: 'Application & data caching with sw-precache'
+perfomance-patterns: 'In memory LRU cache, DNS prefetch, Preact compat alias in production'
+server-side-rendering: 'Yes'
+api: 'Node-hnapi (unofficial)'
+hosting: 'Digital Ocean & Cloudflare'
+authors:
+    - name: 'Chris'
+lighthouse: '91/100'
+interactive-em: '4.7s'
```
Comment (addyosmani) [inline site/_apps/nextjs-hn.txt]: EM TTI according to the Lighthouse report is 4,410ms - We can update this to 4.4s here. 

image

**Comment by addyosmani** [comment]
Great work, @chrisdwheatley!

File: site/_apps/nextjs-hn.txt (hunk 1 above)
Comment (addyosmani) [inline site/_apps/nextjs-hn.txt]: Sigh. That run also failed due to main thread issues. Trying again:

https://www.webpagetest.org/result/170608_DD_b9e141e3eb2598266ac4f7d55a1321ef/

**Comment by chrisdwheatley** [comment]
Thanks @addyosmani, I finally managed to run a test where Lighthouse didn't fail on the main thread issue, see https://www.webpagetest.org/result/170608_YC_J16/.

That test runs on stable Chrome rather than Chrome Beta as I thought that might've been an issue (although it wasn't) but the settings are all the same as the 'Mobile - Faster 3G' settings from the easy tab on WPT.

I've mentioned the issue I'm seeing [here](https://github.com/GoogleChrome/lighthouse/issues/2434#issuecomment-307025305) as it'd be good to understand why I'm seeing it and what can be done about it.

If we get to the bottom of it I'll re-submit with fresh times if required.

**Comment by chrisdwheatley** [comment]
So close yet so far https://www.webpagetest.org/result/170608_X5_6819e7b54214345c826f82327c187600/

**Comment by addyosmani** [comment]
@chrisdwheatley https://github.com/GoogleChrome/lighthouse/pull/2473 on the Lighthouse side should hopefully help here. Once we land that upstream in master it'll require a Lighthouse release bump + WPT updating before we'll see those changes.

**Comment by addyosmani** [comment]
Hey Chris. Per the Ember thread which also had some issues with TTI, we have two choices for moving ahead.

1. Wait for the upstream LH PR to land in a release and get deployed to WPT (few weeks)
2. Use a manual set of trace numbers we look at and use those. We file an issue to update this once LH has the linked to change available.

Preferences?

**Comment by chrisdwheatley** [comment]
Option two would certainly be my preference, which numbers should I use or are the ones I've used right now ok?

**Comment by addyosmani** [comment]
Landing this - I'll update scores manually in master. Thanks again for the submission, @chrisdwheatley!

**Pull request merged**